internal/provider/incident_role_data_source.go

```diff
@@ -0,0 +1,118 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/incident-io/terraform-provider-incident/internal/apischema"
+	"github.com/incident-io/terraform-provider-incident/internal/client"
+)
+
+var (
+	_ datasource.DataSource              = &IncidentRoleDataSource{}
+	_ datasource.DataSourceWithConfigure = &IncidentRoleDataSource{}
+)
+
+func NewIncidentRoleDataSource() datasource.DataSource {
+	return &IncidentRoleDataSource{}
+}
+
+func (i *IncidentRoleDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_incident_role"
+}
+
+type IncidentRoleDataSource struct {
+	client *client.ClientWithResponses
+}
+
+type IncidentRoleDataSourceModel struct {
+	ID           types.String `tfsdk:"id" json:"id"`
+	Name         types.String `tfsdk:"name" json:"name"`
+	Description  types.String `tfsdk:"description" json:"description"`
+	Instructions types.String `tfsdk:"instructions" json:"instructions"`
+	Shortform    types.String `tfsdk:"shortform" json:"shortform"`
+}
+
+func (i *IncidentRoleDataSource) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*IncidentProviderData)
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Data Source Role Configuration",
+			fmt.Sprintf("Expected *client.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	i.client = client.Client
+}
+
+func (i *IncidentRoleDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "This data source provides information about an incident role.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: apischema.Docstring("IncidentRoleV2", "id"),
+				Required:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: apischema.Docstring("IncidentRoleV2", "name"),
+				Computed:            true,
+			},
+			"description": schema.StringAttribute{
```

Comment:
I'm gonna slim this down, I don't think this other information is actually useful to people and it just makes any migrations we might want to do a little harder (and makes the docs less clear imo)